src/ages/ages.ts

```diff
@@ -1,175 +1,63 @@
-        finalBlock: undefined,
-        totalEmission: parseUnits("3333333"),
-        protocolDistribution: undefined,
-      },
-      {
-        id: "age3-epoch2",
-        number: 8,
-        epochName: "epoch2",
-        snapshotBlock: undefined,
-        initialTimestamp: BigNumber.from(new Date("2023-02-01T15:00:00.000Z").getTime() / 1000),
-        initialBlock: undefined,
-        finalTimestamp: BigNumber.from(new Date("2023-03-04T15:00:00.000Z").getTime() / 1000),
-        finalBlock: undefined,
-        totalEmission: parseUnits("3333333"),
-        protocolDistribution: undefined,
-      },
-      {
-        id: "age3-epoch3",
-        number: 9,
-        epochName: "epoch3",
-        snapshotBlock: undefined,
-        initialTimestamp: BigNumber.from(new Date("2023-03-04T15:00:00.000Z").getTime() / 1000),
-        initialBlock: undefined,
-        finalTimestamp: BigNumber.from(new Date("2023-04-07T15:00:00.000Z").getTime() / 1000),
-        finalBlock: undefined,
-        totalEmission: parseUnits("3333334"),
-        protocolDistribution: undefined,
-      },
-    ],
-  },
-];
+const distributions = {
+  age1: ageOneDistribution,
+  age2: ageTwoDistribution,
+  age3: ageThreeDistribution,
+};
 
-export const allEpochs = ages
-  .map((age, ageId) =>
-    age.epochs.map((epoch, epochId) => {
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      const { epochs, ...ageConfig } = age;
-      return { ...epoch, age: age.ageName, ageId, epochId, ageConfig };
-    })
-  )
-  .flat();
+const computeProtocolDistribution = (
+  distribution: typeof agesData[0]["epochs"][0]["protocolDistribution"]
+): ProtocolDistribution | undefined => {
+  if (!distribution) return;
+  const protocolDistribution: ProtocolDistribution = { morphoCompound: BigNumber.from(distribution.morphoCompound) };
+  if ("morphoAave" in distribution) protocolDistribution.morphoAave = BigNumber.from(distribution.morphoAave);
+  return protocolDistribution;
+};
+
+export const ages: AgeConfig[] = agesData.map((data) => {
+  // @ts-ignore
+  const distribution = distributions[data.ageName];
+  const startTimestamp = BigNumber.from(data.startTimestamp);
+  const endTimestamp = BigNumber.from(data.endTimestamp);
+  const epochs = data.epochs.map((epoch) => {
+    const initialTimestamp = BigNumber.from(epoch.initialTimestamp);
+    const finalTimestamp = BigNumber.from(epoch.finalTimestamp);
+    const totalEmission = parseUnits(epoch.totalEmission);
+    const protocolDistribution = computeProtocolDistribution(epoch.protocolDistribution);
+    const finalBlock = epoch.finalBlock ?? undefined;
+    const initialBlock = epoch.initialBlock ?? undefined;
+    const snapshotBlock = epoch.snapshotBlock ?? undefined;
+    return {
+      ...epoch,
+      initialTimestamp,
+      finalTimestamp,
+      totalEmission,
+      protocolDistribution,
+      finalBlock,
+      initialBlock,
+      snapshotBlock,
+    };
```

Comment:
I feel like  we could put that in the  return directly instead of declaring lots of variables